### DISCUSSION TITLE
Task output sometimes cut off

### DISCUSSION THREAD

**Discussion by mmb** [concourse/concourse#5988] [General]
Sometimes the output from a task gets cut off in the middle. The task finishes successfully and the build is green but the output from the container is missing after a certain point.

I was going to open an issue but I have not been able to reliably reproduce it. It is intermittent and there doesn't seem to be a pattern. It doesn't seem to happen more frequently for certain tasks or certain lengths of output.

It's not a UI issue because the output is not in the database. It seems like at some point the container output gets disconnected from whatever is streaming it.

Concourse version is 6.0.0.

**Comment by vito** [comment]
Hmm, maybe check whether any of the `web` nodes were restarted around the time of the cut-off? They'll automatically re-attach but any output that was written while no `web` node was attached will be lost.

**Reply by mmb** [reply]
The concourse web process has been running for a few days and I saw some of these today.

Is it possible the logs being sent back to the server could time out? We do have a pretty busy web node.

**Reply by vito** [reply]
Hmm, it's possible that Garden is dropping windows of output because the `web` node is not quick enough to pull them down. It maintains a buffer and just stops sending things along if the consumer is too slow.

...actually, it looks like [this changed in Feburary](https://github.com/cloudfoundry/garden/commit/2946ade6e5c1a65640a460f8fdbaee8f0c4ba184). But maybe your Concourse cluster hasn't been updated to receive this change. Looks like this shipped in Guardian [1.19.11](https://github.com/cloudfoundry/garden-runc-release/commit/f25eab3c3086082e4f7db3107b1f8814a563b886#diff-25d902c24283ab8cfbac54dfa101ad31) in April, but it's a bit hard to keep tabs on which version shipped with which Concourse.

What Concourse version are you running?

**Reply by mmb** [reply]
We are running Concourse 6.0.0 and ./gdn -v on a worker node shows 90961f153e3c4eccf6c461e9efa5165ac454f47c. Do any of the more recent Concourse releases have this change? We can make the web node bigger too if that would help.

**Reply by vito** [reply]
Yeah, the latest Concourse versions are built with `gdn` 1.19.12, so they should have that change.

Either making the web node bigger or adding another web node might help. The new `gdn` version will prevent logs from being dropped in this case, but there's still the underlying issue of being a slow consumer in the first place. I would look for indicators of that in metrics first though, it's also possible that there's just a ton of output being printed really quickly.
